### PR TITLE
Fix game name glitch

### PR DESCRIPTION
**Pull request by TyrenDe**
First added a clear null check
Second put in a default value
Added a search fallback for suggestions

### CODE REVIEW

File: MixItUp.WPF/Controls/MainControls/ChannelControl.xaml.cs [hunk 1]
```diff
@@ -67,10 +67,14 @@ protected override Task InitializeInternal()
             this.AgeRatingComboBox.ItemsSource = EnumHelper.GetEnumNames<AgeRatingEnum>();
 
             this.StreamTitleTextBox.Text = ChannelSession.Channel.name;
-            if (ChannelSession.Channel.type != null)
+            if (ChannelSession.Channel?.type?.name != null)
```
Comment (SaviorXTanren) [inline MixItUp.WPF/Controls/MainControls/ChannelControl.xaml.cs]: Wouldn't the "type?" fail if there is not Game Type assigned? Maybe add an extra check for null on the type?

File: MixItUp.WPF/Controls/MainControls/ChannelControl.xaml.cs [hunk 2]
```diff
@@ -67,10 +67,14 @@ protected override Task InitializeInternal()
...
             {
                 this.GameNameComboBox.Text = ChannelSession.Channel.type.name;
             }
+            else
+            {
+                this.GameNameComboBox.Text = "<Type a valid game name here>";
```
Comment (SaviorXTanren) [inline MixItUp.WPF/Controls/MainControls/ChannelControl.xaml.cs]: Change this to "Web Show" instead, that'll be a good default

File: MixItUp.WPF/Controls/MainControls/ChannelControl.xaml.cs [hunk 3]
```diff
@@ -227,6 +231,14 @@ private void GameNameTextBox_LostFocus(object sender, System.Windows.RoutedEvent
             if (!string.IsNullOrEmpty(gameName))
             {
                 var games = await ChannelSession.Connection.GetGameTypes(gameName, 10);
+
+                if (games.Count() == 0)
+                {
+                    // If we did a search and it didn't find any relevant games, let's
+                    // get something that is at least actionable.
+                    games = await ChannelSession.Connection.GetGameTypes("game", 10);
```
Comment (SaviorXTanren) [inline MixItUp.WPF/Controls/MainControls/ChannelControl.xaml.cs]: Same thing here, "Web Show"

File: MixItUp.WPF/Controls/MainControls/ChannelControl.xaml.cs (hunk 1 above)
Comment (TyrenDe) [reply MixItUp.WPF/Controls/MainControls/ChannelControl.xaml.cs]: No, this syntax is a bit newer but basically translates this code:
 if (ChannelSession.Channel != null && ChannelSession.Channel.type != null && ChannelSession.Channel.type.name != null) 

So, basically if type is null then the evaluation stops and the rest of the result is null. else, it moves to check name.

Comment (SaviorXTanren) [reply MixItUp.WPF/Controls/MainControls/ChannelControl.xaml.cs]: ooooh, cool :)

**Pull request merged**